src/sentry/rules/processing/delayed_processing.py

```diff
@@ -142,26 +168,60 @@ def get_condition_group_results(
             return None
 
         _, duration = condition_inst.intervals[unique_condition.interval]
-        comparison_interval = condition_inst.intervals[unique_condition.interval][1]
-        comparison_type = (
-            condition_data.get("comparisonType", ComparisonType.COUNT)
-            if condition_data
-            else ComparisonType.COUNT
-        )
+
+        comparison_type = condition_data.get("comparisonType", ComparisonType.COUNT)
+        if comparison_type == ComparisonType.PERCENT:
+            comparison_interval = condition_inst.intervals[
+                condition_data.get("comparisonInterval", "5")
+            ][1]
+        else:
+            comparison_interval = None
+
         result = safe_execute(
             condition_inst.get_rate_bulk,
-            duration,
-            comparison_interval,
-            group_ids,
-            unique_condition.environment_id,
-            comparison_type,
+            duration=duration,
+            group_ids=group_ids,
+            environment_id=unique_condition.environment_id,
+            comparison_type=comparison_type,
+            current_time=current_time,
+            comparison_interval=comparison_interval,
         )
         condition_group_results[unique_condition] = result or {}
     return condition_group_results
 
 
+def _passes_comparison(
+    condition_group_results: dict[UniqueConditionQuery, dict[int, int]],
+    condition_data: EventFrequencyConditionData,
+    group_id: int,
+    environment_id: int,
+) -> bool:
+    """
+    Checks if a specific condition instance has passed.
+    """
+    unique_queries = _generate_unique_queries(condition_data, environment_id)
+    try:
+        query_values = [
+            condition_group_results[unique_query][group_id] for unique_query in unique_queries
+        ]
+    except KeyError as e:
+        logger.exception(
+            "delayed_processing.missing_query_results", extra={"exception": e, "group_id": group_id}
+        )
+        return False
+
+    calculated_value = query_values[0]
+    # If there's a second query we must have a percent comparison condition.
+    if len(query_values) == 2:
+        calculated_value = percent_increase(calculated_value, query_values[1])
+
+    target_value = float(condition_data["value"])
```

Comment:
Previously we converted this to a string then float, not sure if that's important though

Reply:
@ceorourke do you remember why we originally did that?

Reply:
I think it was just to match this https://github.com/getsentry/sentry/blob/master/src/sentry/rules/conditions/event_frequency.py - what's the data type initially? A float?

Reply:
It's a string but I think we can just convert that directly, like [here](https://github.com/getsentry/sentry/blob/master/src/sentry/rules/conditions/event_frequency.py#L156)